crypto/pkcs7/pk7_doit.c

```diff
@@ -1107,8 +1107,10 @@ int PKCS7_signatureVerify(BIO *bio, PKCS7 *p7, PKCS7_SIGNER_INFO *si,
             ret = -1;
             goto err;
         }
-        if (!EVP_VerifyUpdate(mdc_tmp, abuf, alen))
+        if (!EVP_VerifyUpdate(mdc_tmp, abuf, alen)) {
+            OPENSSL_free(abuf);
             goto err;
+        }
```

Comment:
IMO a better fix would be to just move the `OPENSSL_free(abuf)` call below after the `err: `label.

Reply:
Then we would have to change the scope of where `abuf` lives as well. In my own code, I like to keep the declarations of variables as close to the scope that actually uses it to avoid confusion.

Reply:
Updated to move the free operation to the err label